regolith/helpers/u_milestonehelper.py

```diff
@@ -0,0 +1,148 @@
+"""Helper for updating milestones to the projecta collection.
+    It can update the status and due date of a projectum.
+    It can add a new milestone to the projecta collection.
+"""
+import datetime as dt
+import sys
+from dateutil import parser
+from regolith.helpers.basehelper import DbHelperBase
+from regolith.fsclient import _id_key
+from regolith.tools import all_docs_from_collection
+
+
+TARGET_COLL = "projecta"
+
+def subparser(subpi):
+    subpi.add_argument("projectum_id", help="the id of the projectum")
+    subpi.add_argument("--number",
+                        help="number of the milestone to update from numbered list")
+    subpi.add_argument("--due_date",
+                       help="new due date of the milestone in ISO format (YYYY-MM-DD) "
+                            "required to add a new milestone")
+    subpi.add_argument("-s", "--status",
+                       help="initial of the status of the milestone/deliverable: "
+                            "p = proposed, s = started, f = finished, c = cancelled")
+    subpi.add_argument("--name",
+                       help="name of the new milestone. "
+                            "required to add a new milestone")
+    subpi.add_argument("-o", "--objective",
+                       help="objective of the new milestone. "
+                            "required to add a new milestone")
+    subpi.add_argument("-t", "--type",
+                       help="initial of type of the new milestone "
+                            "m = meeting, r = release, p = pull request, o = other")
```

Comment:
build the help string using the ``ALLOWED_TYPES``, sthg like ``f"{*ALLOWED_TYPES.keys()} = {*ALLOWED_TYPES.values()}"``.  This makes it all easier to maintain

Reply:
this won't work, but something like it.  Its fun to make it work, and then it is fixed for ever.

Reply:
you may find ``zip()`` helpful

Reply:
ok, I am using `ALLOWED_TYPES `and `ALLOWED_STATUS` now. 
I couldn't figure out how to properly format it, but I am trying to find a better way. 
does it look bad?
`  -s STATUS, --status STATUS
                        initial of the status of the milestone/deliverable: {'p':
                        'proposed', 's': 'started', 'f': 'finished', 'b': 'back_burner',
                        'pd': 'paused', 'c': 'cancelled'}
  -t TYPE, --type TYPE  initial of type of the new milestone {'m': 'meeting', 'r':
                        'release', 'p': 'pull request', 'o': 'other'}`